src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact (String value, boolean isPrivate) {
+        this.value = value;
+        this.isPrivate = isPrivate;
+    }
+    
+    @Override
+    public String toString() {
+        return value;
+    }
+    
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
+    }
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types.